### PR TITLE
Add lastIndexOf method

### LINKED ISSUE #21236: Add lastIndexOf to lang.string
**Description:**
Please refer https://github.com/ballerina-platform/ballerina-spec/issues/429

**Steps to reproduce:**
ballerina 1.1.0

### LINKED ISSUE #21237: Add lastIndexOf to lang.array
**Description:**
Please refer https://github.com/ballerina-platform/ballerina-spec/issues/429

**Steps to reproduce:**
ballerina 1.1.0

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Add lastIndexOf method to string and array.

Fixes #21236 
Fixes #21237

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java [hunk 1]
```diff
@@ -247,6 +247,23 @@ public void testIndexOf() {
         assertNull(returns[1]);
     }
 
+    @Test
+    public void testLastIndexOf() {
+        BValue[] returns = BRunUtil.invoke(compileResult, "testLastIndexOf");
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java]: Shall we write the tests in Ballerina itself?

Comment (KavinduZoysa) [reply langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java]: Added

File: langlib/lang.string/src/main/java/org/ballerinalang/langlib/string/LastIndexOf.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        orgName = "ballerina", packageName = "lang.string",
+        functionName = "lastIndexOf",
+        args = {@Argument(name = "s", type = TypeKind.STRING),
+                @Argument(name = "substring", type = TypeKind.STRING)},
+        returnType = {@ReturnType(type = TypeKind.UNION)},
+        isPublic = true
+)
+public class LastIndexOf {
+    public static Object lastIndexOf(Strand strand, String value, String subString, long startIndex) {
+        StringUtils.checkForNull(value, subString);
```
Comment (pubudu91) [inline langlib/lang.string/src/main/java/org/ballerinalang/langlib/string/LastIndexOf.java]: Do we need this null check? We should never be getting null values for string values here. If we do get, then it's a bug in either the runtime or the compiler.

File: langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal [hunk 3]
```diff
@@ -150,6 +150,35 @@ function testIndexOf() returns [int?, int?] {
...
+    int? i6 = array.lastIndexOf(arr1);
+
+    int? i7 = array.lastIndexOf(11);
+    int? i8 = array.lastIndexOf("Bar");
+    int? i9 = array.lastIndexOf(12.33);
+    int? i10 = array.lastIndexOf(false);
+    int? i11 = array.lastIndexOf(m2);
+    int? i12 = array.lastIndexOf(arr2);
+
+    if (<int>i1 != 2 && <int>i2 != 5 && <int>i3 != 7 && <int>i4 != 9 && <int>i5 != 11 && <int>i6 != 13 &&
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal]: We should check or instead of and right?

**Pull request merged**